@wrap_exceptions
def io_counters(self):
    try:
        ret = cext.proc_io_counters(self.pid)
    except OSError:
        err = sys.exc_info()[1]
        if (err.errno in ACCESS_DENIED_SET):
            ret = cext.proc_io_counters_2(self.pid)
        else:
            raise
    return _common.pio(*ret)
